File: 프로그래머스/lv0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.java
class Solution {
    public int solution(int[][] lines) {
        int answer = 0;
        int[] arr = new int[200];
        
        for (int i=0; i<lines.length; i++) {
            for (int j=lines[i][0]; j<lines[i][1]; j++) {
                arr[j+100] += 1;
            }
        }
        
        for (int k=0; k<arr.length; k++) {
            if (arr[k] >= 2) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.kt
class Solution {
    fun solution(lines: Array<IntArray>): Int {
        var line = IntArray(200)
        
        for (i in lines) {
            for (j in i[0] + 100 until i[1] + 100) {
                line[j]++
            }
        }
        
        return line.filter { it > 1 }.count()
    }
}